DESIGN.md

```diff
@@ -0,0 +1,40 @@
+# Design
+
+This file intends to provide links to design resources, display the Kiwix colors, as well as presenting accessibility.
+
+## Colors
+
+<img src="https://user-images.githubusercontent.com/22193232/83739760-4ef07080-a656-11ea-8f87-344f48e76324.png" alt="drawing" width="300"/>
+
+The Kiwix colors are black and white with different colored accents. For Kiwix Android, black is accented by different shades of blue:
+
+- ![#000000](https://via.placeholder.com/15/000000/000000?text=+) `#000000`/`black`
```

Comment:
So a part of what I want to see in this doc is true names for these colors

If you haven't watched [this video](https://www.youtube.com/watch?v=Owkf8DhAOSo) already I highly recommend it

[This part](https://youtu.be/Owkf8DhAOSo?t=1058) particularly about naming colors is something I want to see as `color_primary` is a pretty terrible name which is my fault.

We also need to be aware of night mode with this document.

Reply:
It probably would encompass changing the colors in the project, we could make a ticket "align colors with design doc" or it might be simpler for you to do them in tandem?

We will have to update the theme also as I believe I was overriding color resources in values-night to achieve the dark mode functionality.

black can probably be left as black, it describes it well and isn't ambiguous or at least I am thinking of the singular value `#000000`.
blue raises a more ambiguous question as blue is a wider spectrum, we would want a name that can distinguish it from other blues should we include blues, and we want to avoid a scenario of `kiwix_blue_1_800`, `kiwix_blue_2_800` as that won't help anyone understand the color system. Code words can be adopted eg `sky` `water` or probably something better with a punchier name. What really matters is consistent usage and the ability to - at a glance - be able to distinguish the color

Reply:
I found a site for translating hex to color names [1] and got  `#1565c0` = `denim_800`. Don't know how accurate that is. Then `#2196F3` = `denim_500` and `#42a5f5` = `denim_400`. 

I found this an article [2] suggesting functionality should be included in the name of a color e.x. `button_primary`=`denim_800`. Feels like that can be a little bloaty though, or annoying in that we'd have to create a new color for each new view. 
 
[1] - [http://chir.ag/projects/name-that-color/#1565C0](http://chir.ag/projects/name-that-color/#1565C0)
[2] - [https://www.novatec-gmbh.de/en/blog/name-android-colors-palettes/](https://www.novatec-gmbh.de/en/blog/name-android-colors-palettes/)

Reply:
Step 1 is to define all the colors with their actual names first and that site seems a sensible enough way of doing it.

If we need to add a layer of indirection between how our colors are used we can but right now there is no need, we rarely specify the color of an element, we use sensible defaults by assigning our theme attributes and in the scenario where we would want a button to deviate for our theme it would either get its own theme/style/overlay

Reply:
The reason I suggested `kiwix_blue` is that it seems like we are mainly using three shades of blue across the app (different shades of "kiwix_blue"). If we were to include a different blue I think it should be named e.x. `sky` as it is not the "brand" color. I think `kiwix_blue` is the most descriptive name if we are not to name the color after a functionality. 

I immediately think of the color of the big blue button at the start screen when reading `kiwix_blue`. However, I think it is a little difficult to know which shade (between 800, 500, 400) is the primary `kiwix_blue` though.

Reply:
That is a slightly wrong way of thinking, that button is not blue or more accurately not always the same blue. It is the attribute `primaryColor` (I think, I'd have to look) and that is all we care about because that color adapts to dark mode.

`kiwix_blue` can work and then I guess if a new blue were to ever come in we could name it something distinct that contains neither `kiwix` or `blue`

Reply:
I got the idea from the video where they used `brand_blue` as an example of a color name. Anyway, the important thing is to rename the colors and remain consistent. If it is ok, I'll change the color names (in the project) in this ticket.

> We will have to update the theme also as I believe I was overriding color resources in values-night to achieve the dark mode functionality.

I'll look into this as well.